src/library/scala/collection/SeqView.scala

```diff
@@ -112,6 +114,36 @@ object SeqView {
     @throws[IndexOutOfBoundsException]
     def apply(i: Int) = underlying.apply(i)
   }
+
+  @SerialVersionUID(3L)
+  class Sorted[A, B >: A](underlying: SomeSeqOps[A])(implicit ord: Ordering[B]) extends SeqView[A] {
+    private[this] lazy val _sorted = {
+      val len = underlying.length
+      new SeqView.Id(
+        if (len == 0) Nil
+        else if (len == 1) List(underlying.head)
+        else {
+          val arr = new Array[Any](len) // Array[Any] =:= Array[AnyRef]
+          underlying.copyToArray(arr)
+          java.util.Arrays.sort(arr.asInstanceOf[Array[AnyRef]], ord.asInstanceOf[Ordering[AnyRef]])
+          // casting the Array[AnyRef] to Array[A] and creating an ArraySeq from it
+          // is safe because:
+          //   - the ArraySeq is immutable, and items that are not of type A
+          //     cannot be added to it
+          //   - we know it only contains items of type A (and if this collection
+          //     contains items of another type, we'd get a CCE anyway)
+          //   - the cast doesn't actually do anything in the runtime because the
+          //     type of A is not known and Array[_] is Array[AnyRef]
+          immutable.ArraySeq.unsafeWrapArray(arr.asInstanceOf[Array[A]])
+        }
+      )
+    }
+
+    def apply(i: Int): A = _sorted.apply(i)
+    def length: Int = underlying.length
+    def iterator: Iterator[A] = Iterator.empty ++ _sorted.iterator // very lazy
+    override def knownSize: Int = underlying.knownSize
```

Comment:
Maybe have
```scala
override def sorted(implicit ord: Ordering[A]): SeqView[A] =
  if (ord == this.ord) this else new Sorted(underlying)(ord)
```

At first I thought "well, if you do `.sorted.sorted` you deserve something bad" however I thought it could happen innocently if code is separated...
```scala
//somewhere in code
val myView = (??? : SeqView[Int]).sorted
takesAnyView(myView)

// somewhere else..
def takesAnyView(seqView: SeqView[A]): Unit = seqView.sorted.foreach(println)
```

Reply:
valid. that's easy to add